### PR TITLE
Load - Logger instead of Exceptions

### PR DESCRIPTION
**Pull request by hsorby**
My attempt at adding a Logger instead of using exceptions.

### CODE REVIEW

File: src/api/libcellml/logger.h [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2015 University of Auckland
```
Comment (nickerso) [inline src/api/libcellml/logger.h]: 2016 :)

File: src/api/libcellml/types.h [hunk 2]
```diff
@@ -31,6 +32,14 @@ class Units; /**< Forward declaration of Units class. */
 typedef std::shared_ptr<Units> UnitsPtr; /**< Type definition for shared units pointer. */
 class Variable; /**< Forward declaration of Variable class. */
 typedef std::shared_ptr<Variable> VariablePtr; /**< Type definition for shared variable pointer. */
+class XmlNode; /**< Forward declaration of XmlNode class. */
+typedef std::shared_ptr<XmlNode> XmlNodePtr; /**< Type definition for shared XML node pointer. */
```
Comment (dladd) [inline src/api/libcellml/types.h:36]: I'm not quite sure why the XML classes and pointers are being exposed in the libcellml types? Or is this a case of your branch being a commit or two behind mine?

File: src/api/libcellml/logger.h (hunk 1 above)
Comment (dladd) [inline src/api/libcellml/logger.h]: We might need to do a library-wide search and replace for this after this pull request. And maybe also update CellML 1.2 to 2.0 in the same tidy-up pull.

File: src/api/libcellml/types.h (hunk 2 above)
Comment (hsorby) [inline src/api/libcellml/types.h:36]: They are part of the interface albeit the private interface nevertheless we need to forward declare these objects.  It's an improvement over including non-API headers in the API header files.

**Pull request merged**